### PR TITLE
PurchaseTester: Add new UI to configure internal proxy behavior

### PR DESCRIPTION
**Pull request by tonidero**
### Description
This PR adds a new button to the `ConfigureFragment`, `LoginFragment` and `OverviewFragment`. This will open a bottom sheet where we can modify the behavior of our internal proxy. This will help with testing offline entitlements and trusted entitlements. Note that this UI will only be helpful when using our internal proxy. When using other proxies, the requests will fail and the UI will display the corresponding error

https://github.com/RevenueCat/purchases-android/assets/808417/4455da06-0c44-40e2-b3a3-f0aa572f294b

### CODE REVIEW

File: examples/purchase-tester/src/main/java/com/revenuecat/purchasetester/ConfigureFragment.kt [hunk 1]
```diff
@@ -56,6 +58,26 @@ class ConfigureFragment : Fragment() {
             }.collect()
         }
 
+        @Suppress("EmptyFunctionBlock")
+        binding.proxyUrlInput.addTextChangedListener(object : TextWatcher {
```
Comment (tonidero) [inline examples/purchase-tester/src/main/java/com/revenuecat/purchasetester/ConfigureFragment.kt:62]: I added this so we could configure the proxy setting from before the SDK was configured. This means we will update the property on every change, which is probably fine.

**Comment by NachoSoto** [comment]
This looks amazing 😍 
We could do something similar in iOS. I did this for macOS only because it was much easier as a separate window.

I'll let somebody else review this cause I know virtually nothing about Android UI.

File: examples/purchase-tester/src/main/java/com/revenuecat/purchasetester/ConfigureFragment.kt (hunk 1 above)
Comment (aboedo) [reply examples/purchase-tester/src/main/java/com/revenuecat/purchasetester/ConfigureFragment.kt:62]: it's fine for a sample app. Would listening to `onFocusChange` also do the trick, but without the intermediate calls while editing?

**Review by aboedo — APPROVED** [review]
love it!

File: examples/purchase-tester/src/main/java/com/revenuecat/purchasetester/ConfigureFragment.kt (hunk 1 above)
Comment (tonidero) [reply examples/purchase-tester/src/main/java/com/revenuecat/purchasetester/ConfigureFragment.kt:62]: There are some situations where the edittext doesn't lose focus, causing the proxy url to not be updated in time. For example, if you are editting the proxy url edittext and then click a button, the focus isn't lost. Updating on text change was the quickest alternative I could find, even though it's clearly not optimal

**Comment by tonidero** [comment]
Merging this for now.

**Pull request merged**